src/GraphQLParser/Visitors/SDLWriter.cs

```diff
@@ -0,0 +1,765 @@
+
+namespace GraphQLParser.Visitors
+{
+    /// <summary>
+    /// Prints AST into the provided <see cref="TextWriter"/> as a SDL document.
+    /// </summary>
+    /// <typeparam name="TContext">Type of the context object passed into all VisitXXX methods.</typeparam>
+    public class SDLWriter<TContext> : DefaultNodeVisitor<TContext>
+        where TContext : IWriteContext
+    {
+        /// <inheritdoc/>
+        public override async ValueTask VisitDocument(GraphQLDocument document, TContext context)
+        {
+            if (document.Definitions?.Count > 0)
+            {
+                for (int i = 0; i < document.Definitions.Count; ++i)
+                {
+                    await Visit(document.Definitions[i], context);
+
+                    if (i < document.Definitions.Count - 1)
+                    {
+                        await context.Writer.WriteLineAsync();
+                    }
+                }
+            }
+        }
+
+        /// <inheritdoc/>
+        public override async ValueTask VisitComment(GraphQLComment comment, TContext context)
+        {
+            int level = GetLevel(context);
+
+            bool needStartNewLine = true;
+            int length = comment.Text.Span.Length;
+            for (int i = 0; i < length; ++i)
+            {
+                if (needStartNewLine)
+                {
+                    await WriteIndent(context, level);
+                    await context.Writer.WriteAsync('#');
+                    needStartNewLine = false;
+                }
+
+                char current = comment.Text.Span[i];
+                switch (current)
+                {
+                    case '\r':
+                        break;
+
+                    case '\n':
+                        await context.Writer.WriteLineAsync();
+                        needStartNewLine = true;
+                        break;
+
+                    default:
+                        await context.Writer.WriteAsync(current);
+                        break;
+                }
+            }
+
+            await context.Writer.WriteLineAsync();
+        }
+
+        /// <inheritdoc/>
+        public override async ValueTask VisitDescription(GraphQLDescription description, TContext context)
+        {
+            int level = GetLevel(context);
+
+            await WriteIndent(context, level);
+            await context.Writer.WriteAsync("\"\"\"");
+            await context.Writer.WriteLineAsync();
+
+            bool needStartNewLine = true;
+            int length = description.Value.Span.Length;
+            for (int i = 0; i < length; ++i)
+            {
+                if (needStartNewLine)
+                {
+                    await WriteIndent(context, level);
+                    needStartNewLine = false;
+                }
+
+                char current = description.Value.Span[i];
+                switch (current)
+                {
+                    case '\r':
+                        break;
```

Comment:
Note: with this logic, description strings encoded with CRLF will result in strings being encoded as LF after being decoded.  I suggest adding a comment within the method that this change may take place -- or better yet, encode strings containing CRs as strings instead of blockstrings.  Tests should be added either way.

Reply:
> strings instead of blockstrings

I'm still confused about string vs blockstring, can you give an example?

By the way I want to rewrite both methods for comment and description node. Now I added the `Slice` call to get around the problem:
```
CS4012	Parameters or locals of type 'ReadOnlySpan<char>' cannot be declared in async methods or async lambda expressions.
```

In addition, I do not really like this char-by-char transmission. At first I wrote a method that finds the chain of non-`\r\n` chars and writes those spans into writer, but it was rather complicated so I swiched to char-by-char.